snapcraft/storeapi/errors.py

```diff
@@ -111,28 +111,63 @@ def __init__(self, exception):
         super().__init__(message=message)
 
 
-class SnapNotFoundError(StoreError):
+class SnapNotFoundError(SnapcraftException):
+    def __init__(
+        self,
+        *,
+        snap_name: str = "",
+        snap_id: Optional[str] = None,
+        channel: Optional[str] = None,
+        arch: Optional[str] = None,
+    ):
+        super().__init__()
 
-    __FMT_ARCH_CHANNEL = (
-        "Snap {name!r} for {arch!r} cannot be found in the {channel!r} channel."
-    )
-    __FMT_CHANNEL = "Snap {name!r} was not found in the {channel!r} channel."
-    __FMT_SERIES_ARCH = "Snap {name!r} for {arch!r} was not found in {series!r} series."
-    __FMT_SERIES = "Snap {name!r} was not found in {series!r} series."
+        # Defaulting snap_name to "" to support the one case we have that
+        # makes use of snap_id.
+        if snap_name == "" and snap_id is None:
+            raise RuntimeError("Both 'snap_name' and 'snap_id' cannot be None.")
+
+        self._snap_name = snap_name
+        self._snap_id = snap_id
+        self._channel = channel
+        self._arch = arch
+
+    def get_brief(self) -> str:
+        if self._snap_id:
+            # This is legacy.
+            brief = f"Cannot find snap with snap_id {self._snap_id!r}."
+        elif self._channel and self._arch:
+            brief = f"Snap {self._snap_name!r} for architecture {self._arch!r} was not found on channel {self._channel!r}."
+        elif self._channel:
+            brief = (
+                f"Snap {self._snap_name!r} was not found on channel {self._channel!r}."
+            )
+        elif self._arch:
+            brief = f"Snap {self._snap_name!r} for architecture {self._arch!r} was not found."
+        else:
+            brief = f"Snap {self._snap_name!r} was not found."
+
+        return brief
 
-    fmt = "Snap {name!r} was not found."
+    def get_resolution(self) -> str:
+        # All new APIs use snap_name.
+        if self._snap_id:
+            snap_identifier = self._snap_id
+        else:
+            snap_identifier = self._snap_name
 
-    def __init__(self, name, channel=None, arch=None, series=None):
-        if channel and arch:
-            self.fmt = self.__FMT_ARCH_CHANNEL
-        elif channel:
-            self.fmt = self.__FMT_CHANNEL
-        elif series and arch:
-            self.fmt = self.__FMT_SERIES_ARCH
-        elif series:
-            self.fmt = self.__FMT_SERIES
+        resolution = f"Ensure you have proper access rights for {snap_identifier!r}."
```

Comment:
Just a though: I'm thinking I tend to use "verify" but here you use "ensure".  We should probably have a template for the errors messages with the common language to improve consistency, whichever way.

Reply:
This is master (includes docstrings, method names and code comments though):
```
sergiusens@gotham:~/projects/snapcore/snapcraft$ grep -i -R ensure snapcraft  | wc -l
134
sergiusens@gotham:~/projects/snapcore/snapcraft$ grep -i -R verify snapcraft  | wc -l
57
```
I am pretty sure that ensure comes from a design review way back then.

Reply:
Sounds good!  I'll try to remember :D